### PR TITLE
Enable USB on ESP32-P4

### PR DESCRIPTION
**Pull request by tannewt**

### CODE REVIEW

**Comment by arturo182** [comment]
Hurray. Are there any requirements regarding silicon revision?

**Comment by tannewt** [comment]
I don't think so. I have chip version 0.1.

**Comment by dhalbert** [comment]
There are TinyUSB build errors, it looks like.

File: supervisor/shared/usb/tusb_config.h [hunk 1]
```diff
@@ -89,6 +89,12 @@ extern "C" {
 #endif
 #endif
 
+// Use DMA with the USB peripheral.
+#ifdef CONFIG_IDF_TARGET_ESP32P4
+#define CFG_TUD_DWC2_DMA_ENABLE (1)
+#define CFG_TUH_DWC2_DMA_ENABLE (1)
```
Comment (hathach) [inline supervisor/shared/usb/tusb_config.h:95]: DMA can be enabled for S2/S3 as well as they share pretty much same usbip. 
- For P4, it use DCache and will need memory align 64 (variable must also occupies the whole cache line 64). 
- For S2/S3, aligne 4 bytes are good enough.

**Comment by tannewt** [comment]
`hcd_int_handler` errors will be fixed by https://github.com/hathach/tinyusb/pull/2895

**Comment by tannewt** [comment]
This is passing! Please take a look.

**Pull request merged**